dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        var requestBody = Json.Serialize(new ImageGenerationRequest
+        {
+            Prompt = description,
+            Size = $"{width}x{height}",
+            Count = 1
+        });
+
+        var result = await this.GetImageGenerationAsync(this._endpoint, requestBody, cancellationToken).ConfigureAwait(false);
+        if (result.Result == null)
+        {
+            throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Response not contains result");
+        }
+
+        return result.Result.Images[0].Url;
+    }
+
+    private async Task<AzureImageGenerationResponse> GetImageGenerationAsync(string url, string? requestBody, CancellationToken cancellationToken = default)
+    {
+        HttpResponseMessage? response = null;
+        var retryCount = 0;
+        try
+        {
+            using (var content = new StringContent(requestBody, Encoding.UTF8, "application/json"))
+            {
+                response = await this.ExecuteRequestAsync(url, HttpMethod.Post, content, cancellationToken).ConfigureAwait(false);
+            }
+
+            var responseJson = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            var result = this.JsonDeserialize<AzureImageGenerationResponse>(responseJson);
+
+            if (!response.Headers.TryGetValues("operation-location", out var locationValues) || string.IsNullOrWhiteSpace(locationValues.FirstOrDefault()))
+            {
+                throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Not found operation-location");
+            }
+
+            var operationLocation = locationValues.First();
+            while (true)
+            {
+                if (result.Status.Equals(SucceededStatus, StringComparison.OrdinalIgnoreCase))
+                {
+                    return result;
+                }
+
+                if (this._maxRetryCount == retryCount)
+                {
+                    throw new AIException(AIException.ErrorCodes.RequestTimeout, "Reached maximum retry attempts");
+                }
+
+                if (response.Headers.TryGetValues("retry-after", out var afterValues) && long.TryParse(afterValues.FirstOrDefault(), out var after))
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(after), cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(this._defaultRetryAfter), cancellationToken).ConfigureAwait(false);
+                }
+
+                response = await this.ExecuteRequestAsync(operationLocation, HttpMethod.Get, null, cancellationToken).ConfigureAwait(false);
+                responseJson = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                result = this.JsonDeserialize<AzureImageGenerationResponse>(responseJson);
+
+                // increase retry count
+                retryCount++;
```

Comment:
I'm curious if we need to do this retry or not. The kernel supports adding this handling by default or custom httpClients can handle themselves.

Reply:
There are some differences in the retry process here compared to httpClient. 

Azure's Dall E API follows the following process:
First, calling the submit API generates a task, and the response header includes the operation-location indicating the address to query for the task result. 
Then, it is necessary to continuously attempt to check the generated result from the operation-location. 
The ongoing response header also specifies a Retry-After parameter indicating the delay for retrying until the final result state changes to succeeded, which signifies a successful request.

Therefore, some retry steps and limitations have been added here to avoid getting stuck in an infinite loop.

The currently known statuses include notRunning, running, and succeeded. 
So far, no failed cases have been encountered, and it cannot be guaranteed that all tasks will be successful. 
Perhaps it would be advisable to only perform periodic queries during the notRunning and running states?
Any better suggestions are welcome.